ls1intum/Artemis
# Allow downloading binary files from plagiarism inspector

### Checklist

### Motivation and Context


When checking two repositories for plagiarism, the comparison view allows selecting non-text files. They get rendered as text and eventually crash the browser. 

### Description

The plagiarism comparison view now probes the content type of the selected file. If it's not a text file, it won't be rendered. Instead the user is able to download the file.

### Steps for Testing


1. Upload a binary file to a repository (or use ts1 -> Joscha Henningsens Test Course -> Plagiatstest)
2. Check for plagiarism and select the uploaded file. (On my course there is a mp3 file in the repository of test user 1)
3. The file should not be loaded. 
4. Test if downloading the file works as intended.


### Screenshots



before | after 
--- | ---
![image](https://user-images.githubusercontent.com/44805696/118407483-37e93980-b681-11eb-8d94-74815d388dd4.png) | ![image](https://user-images.githubusercontent.com/44805696/118409227-9ca89200-b689-11eb-982b-c9014d8d0ebb.png)
